### PR TITLE
 systrayhelper: 0.0.4 -> 0.0.5

### PR DESCRIPTION
**Pull request by cryptix**
###### Motivation for this change

updates to the latest release v0.0.5 which includes some fixes and dependency updates.

also includes a checkPhase using xvfb_run to spawn go test, which starts i3 and the helper and then uses xdotool to send clicks back to the helper

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Assured whether relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

**Comment by rasendubi** [comment]
Is this a WIP? Otherwise, you need remove commented out code / todos (`don't forget ...`)

**Comment by cryptix** [comment]
@rasendubi: indeed an oversight, thanks! 

I'll try to add the ffmpeg record mechanics in pre/postCheckHook so that I can use them through an override from the source.

**Comment by cryptix** [comment]
@rasendubi sorry, yes/no. this mostly got dragged behind. I pushed a rebased and cleaned up version.

I moved the commented stuff to export test video recordings into the override.nix in the source of the project to unclutter this one.

Also removed the deps.nix file since this uses vendored dependencies.

File: pkgs/tools/misc/systrayhelper/default.nix [hunk 1]
```diff
@@ -1,21 +1,27 @@
-{ stdenv, pkgconfig, libappindicator-gtk3, buildGoPackage, fetchFromGitHub }:
+{ stdenv, pkgconfig, libappindicator-gtk3, buildGoPackage, fetchFromGitHub
+
+# for tests
+, dbus
+, i3
+, xdotool
+#, ffmpeg
```
Comment (rasendubi) [inline pkgs/tools/misc/systrayhelper/default.nix]: Commented out code?

File: pkgs/tools/misc/systrayhelper/default.nix [hunk 2]
```diff
@@ -29,6 +35,20 @@ buildGoPackage rec {
   nativeBuildInputs = [ pkgconfig libappindicator-gtk3 ];
   buildInputs = [ libappindicator-gtk3 ];
 
+  enableParallelBuilding = true;
+
+  checkInputs = [ xvfb_run dbus i3 xdotool ];
+  doCheck = stdenv.isLinux;
+  # use the override.nix from the source to extract and watch tests
+  checkPhase = ''
+    export PATH=$PATH:$(pwd)/go/bin:${i3}/bin:${xdotool}/bin
```
Comment (rasendubi) [inline pkgs/tools/misc/systrayhelper/default.nix]: Are `i3` and `xdotool` necessary here? I think `checkInputs` should add them to `$PATH` automatically.

**Review by rasendubi — COMMENTED** [review]
Minor comments. Otherwise, it looks good.

When you think PR is ready for the final review and merge, remove `[WIP]` from the title.

**Comment by cryptix** [comment]
updated to respect review comments and rebased on master.

**Comment by rasendubi** [comment]
@GrahamcOfBorg build systrayhelper

**Comment by rasendubi** [comment]
nix-review fails for me with the following error:
```
builder for '/nix/store/pbyhaj1ls312xlmvxgxljls5n7m2bwcv-systrayhelper-0.0.5.drv' failed with exit code 1; last 10 log lines:
      xvfb_test.go:32: found i3-msg here: /nix/store/nnskv7cmcv35nvhsac5pcjfddqd6i2lr-i3-4.16.1/bin/i3-msg
      xvfb_test.go:32: found xdotool here: /nix/store/ivkzr71ck6in70p5k8gr0mqm8zbxdi81-xdotool-3.20160805.1/bin/xdotool
      require.go:794: 
          	Error Trace:	xvfb_test.go:31
          	Error:      	Received unexpected error:
          	            	exec: "systrayhelper": executable file not found in $PATH
          	Test:       	TestClicking
          	Messages:   	did not find systrayhelper
  FAIL
  FAIL	github.com/ssbc/systrayhelper/test	0.003s
cannot build derivation '/nix/store/f3fyzpn9y8j0h1kblx5mrhsfhvfb2749-env.drv': 1 dependencies couldn't be built
```

I think `export PATH=$PATH:$(pwd)/go/bin` is still needed in `checkPhase` (without `i3` and `xdotool`)

**Comment by doronbehar** [comment]
I'm closing as CI fails and OP seems to not care as for making this update work and to much time has passed.

**Pull request closed without merging**